cmd/collector/app/zipkin/json.go

```diff
@@ -248,49 +249,20 @@ func float64bytes(float float64) []byte {
 }
 
 func parseIpv4(str string) (int32, error) {
-	// TODO use net.ParseIP
-	segments := strings.Split(str, ".")
-	if len(segments) == 1 {
+	if str == "" {
 		return 0, nil
 	}
 
+	ip := net.ParseIP(str).To4()
+	if ip == nil {
```

Comment:
note that zipkin model currently supports ipv6 as well

Reply:
At the moment we construct that as `Ipv6: []byte(e.IPv6)` Which I guess is not ideal.